Casks/q/qtspim.rb

```diff
@@ -8,11 +8,15 @@
   homepage "https://spimsimulator.sourceforge.net/"
 
   livecheck do
-    url "https://sourceforge.net/projects/spimsimulator/files/"
-    regex(/QtSpim[._-]v?(\d+\.\d+\.\d+)[._-]mac\.(?:mpkg|pkg)(?:\.zip)?/i)
+    url :url
+    regex(%r{url=.*?/QtSpim[._-]v?(\d+(?:\.\d+)+)[._-]mac\.(?:mpkg|pkg)(?:\.zip)?}i)
```

Comment:
```suggestion
    regex(%r{url=.*?/QtSpim[._-]v?(\d+(?:\.\d+)+)[._-]mac\.(?:m?pkg(?:\.zip)?|dmg)}i)
```

The most recent files for macOS use a format like `QtSpim_9.1.24_mac.mpkg.zip` but there's also older releases that use a format like `QtSpim_9.1_mac.dmg`. We can set this up in a way that will match both `.mpkg.zip` and `.dmg` while also allowing for `.pkg` (in case that appears in the future).